### PR TITLE
feat: Install Pydantic

### PR DESCRIPTION
**Pull request by glevco**
This PR simply installs the pydantic lib and sets some configuration. Please read https://github.com/HathorNetwork/hathor-core/pull/519.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -97,7 +98,10 @@ warn_unused_ignores = true
 namespace_packages = true
 show_error_codes = true
 show_error_context = true
-plugins = ["mypy_zope:plugin"]
+plugins = [
+    "mypy_zope:plugin",
+    "pydantic.mypy"
```
Comment (glevco) [inline pyproject.toml:103]: Fix this, the plugin is not working

File: hathor/utils/pydantic.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from pydantic import BaseModel as PydanticBaseModel, Extra
+from pydantic.generics import GenericModel as PydanticGenericModel
+
+
+class BaseModel(PydanticBaseModel):
```
Comment (msbrogli) [inline hathor/utils/pydantic.py:5]: Missing docstring explaining what it is. Maybe add some guidance for whoever is gonna use it in the future.

File: hathor/utils/pydantic.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    class Config:
+        allow_mutation = False
+        extra = Extra.forbid
+
+
+class GenericModel(PydanticGenericModel):
```
Comment (msbrogli) [inline hathor/utils/pydantic.py:11]: Missing docstring explaining what it is. Maybe add some guidance for whoever is gonna use it in the future.

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: hathor/utils/pydantic.py (hunk 2 above)
Comment (glevco) [reply hathor/utils/pydantic.py:19]: Done!

File: hathor/utils/pydantic.py (hunk 3 above)
Comment (glevco) [reply hathor/utils/pydantic.py:32]: Done!

File: pyproject.toml (hunk 1 above)
Comment (glevco) [reply pyproject.toml]: For some reason it was conflicting with `mypy_zope:plugin`. I found out that inverting the order works.

**Review by jansegre — APPROVED** [review]

**Review by msbrogli — APPROVED** [review]

File: hathor/utils/pydantic.py (hunk 3 above)
Comment (msbrogli) [reply hathor/utils/pydantic.py:32]: Thanks!

File: hathor/utils/pydantic.py (hunk 2 above)
Comment (msbrogli) [reply hathor/utils/pydantic.py:19]: Thanks!

**Pull request merged**